src/Digdir.Domain.Dialogporten.Infrastructure/Altinn/Authorization/DecisionRequestHelper.cs

```diff
@@ -178,9 +194,19 @@ private static XacmlJsonCategory CreateResourceCategory(string id, string servic
             }
         }
 
-        if (subResource is not null)
+        if (authorizationAttribute is not null)
         {
-            attributes.Add(new XacmlJsonAttribute { AttributeId = AttributeIdSubResource, Value = subResource });
+            var resourceAttributesFromAuthorizationAttribute = GetResourceAttributesForAuthorizationAttribute(authorizationAttribute);
+
+            // If we get either urn:altinn:app/urn:altinn:org or urn:altinn:resource attributes, this should
+            // be considered overrides that should be used instead of the default resource attributes.
+            if (resourceAttributesFromAuthorizationAttribute.Any(x => x.AttributeId is AttributeIdApp or AttributeIdResource))
```

Comment:
Should this check also include `urn:altinn:org`, as the per the comment?

Reply:
Added for clarity, but in practice a `urn:altinn:app` XACML resource attribute must always be accompanied by a `urn:altinn:org`, so just checking for either works here.